### PR TITLE
lambda improvements and buffering

### PR DESCRIPTION
**Pull request by latifuysal**
## Summary

I used Telemetry API instead of Logs API, because Logs API is not recommended any more.
Improved error handling. 
Added buffering to pushers.

Fixes # [LS-395](https://edgedelta.atlassian.net/browse/LS-395)

## Testing

Not tested yet. It might take some time to test everything.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -20,77 +19,133 @@ var (
...
 				log.Printf("Error during Next Event call: %v", err)
+				continue
+			}
+			log.Printf("Received next event type: %s", eventType)
+			switch eventType {
+			case lambda.Invoke:
+				invokeEvent, err := lambda.GetInvokeEvent(eventBody)
+				if err != nil {
+					log.Printf("Failed to parse Invoke event, err: %v", err)
+				}else{
```
Comment (ozcelgozde) [inline main.go:132]: nit: space before else

File: cfg/config.go [hunk 2]
```diff
@@ -64,37 +68,37 @@ func GetConfigAndValidate() (*Config, error) {
...
 	} else {
-		config.BufferSize = 100
+		config.BufferSize = 10 * 1000 * 1000
 	}
 
-	retryTimeout := os.Getenv("ED_RETRY_TIMEOUT")
-	if retryTimeout != "" {
-		if i, err := strconv.ParseInt(retryTimeout, 10, 0); err == nil {
-			config.RetryTimeout = time.Duration(i)
+	pushTimeout := os.Getenv("ED_PUSH_TIMEOUT")
```
Comment (ozcelgozde) [inline cfg/config.go:82]: we need to update existing lambdas deployed with this extension and also update the documentation.

File: main.go (hunk 1 above)
Comment (latifuysal) [reply main.go]: fixed

**Comment by latifuysal** [comment]
@tuncerkaplankiran  I tested this on us-west-2 staging with the flog lambda: https://us-west-2.console.aws.amazon.com/lambda/home?region=us-west-2#/functions/flog-log-generator
I will test some more, and update README. We can sync and deploy it later.

**Pull request merged**